runtime/builtin_types_test.go

```diff
@@ -23,6 +23,32 @@ import (
 	"testing"
 )
 
+func TestBuiltinDelAttr(t *testing.T) {
+	f := NewRootFrame()
+	delattr := mustNotRaise(Builtins.GetItemString(NewRootFrame(), "delattr"))
+	fooType := newTestClass("Foo", []*Type{ObjectType}, newStringDict(map[string]*Object{}))
+	fooForDelAttr := newObject(fooType)
+	SetAttr(f, fooForDelAttr, NewStr("bar"), None)
+	fun := wrapFuncForTest(func(f *Frame, args ...*Object) (*Object, *BaseException) {
+		_, raised := delattr.Call(f, args, nil)
+		if raised != nil {
+			return nil, raised
+		}
+		return GetAttr(f, args[0], toStrUnsafe(args[1]), nil)
```

Comment:
I am getting strange results, I lost somewhere. please take a look (For first test case).
It is throwing, even if I return nil in exception.

```
% git diff                                                                                                                     [add_delattr]$
diff --git a/runtime/builtin_types_test.go b/runtime/builtin_types_test.go
index 53eeca8..f1f3aee 100644
--- a/runtime/builtin_types_test.go
+++ b/runtime/builtin_types_test.go
@@ -26,18 +26,24 @@ import (
 func TestBuiltinDelAttr(t *testing.T) {
        f := NewRootFrame()
        delattr := mustNotRaise(Builtins.GetItemString(NewRootFrame(), "delattr"))
-       fooType := newTestClass("Foo", []*Type{ObjectType}, newStringDict(map[string]*Object{}))
+       fooType := newTestClass("Foo", []*Type{ObjectType}, NewDict())
        fooForDelAttr := newObject(fooType)
-       SetAttr(f, fooForDelAttr, NewStr("bar"), None)
+       mustNotRaise(nil, SetAttr(f, fooForDelAttr, NewStr("bar"), None))
        fun := wrapFuncForTest(func(f *Frame, args ...*Object) (*Object, *BaseException) {
-               _, raised := delattr.Call(f, args, nil)
+               result, raised := delattr.Call(f, args, nil)
                if raised != nil {
                        return nil, raised
                }
-               return GetAttr(f, args[0], toStrUnsafe(args[1]), nil)
+               val, raised := GetAttr(f, args[0], toStrUnsafe(args[1]), nil)
+               if raised != nil {
+                       fmt.Println(22222222, args, result, val == nil, raised)
+                       return newTestTuple(result, val == nil).ToObject(), nil
+               }
+               fmt.Println(333333333, args)
+               return nil, f.RaiseType(AttributeErrorType, fmt.Sprintf("'delattr' failed to remove '%s', got '%s' instead", args[1], val))
        })
        cases := []invokeTestCase{
-               {args: wrapArgs(fooForDelAttr, "bar"), wantExc: mustCreateException(AttributeErrorType, "'Foo' object has no attribute 'bar'")},
+               {args: wrapArgs(fooForDelAttr, "bar"), want: newTestTuple(None, True.ToObject()).ToObject()},
                {args: wrapArgs(fooForDelAttr, "baz"), wantExc: mustCreateException(AttributeErrorType, "'Foo' object has no attribute 'baz'")},
                {args: wrapArgs(fooForDelAttr), wantExc: mustCreateException(TypeErrorType, "'delattr' requires 2 arguments")},
                {args: wrapArgs(fooForDelAttr, "foo", "bar"), wantExc: mustCreateException(TypeErrorType, "'delattr' requires 2 arguments")},

/Users/name/repo/grumpy% make build/runtime.pass
22222222 [<Foo object at 0xc420425f20> 'bar'] None true AttributeError('\'Foo\' object has no attribute \'bar\'',)
--- FAIL: TestBuiltinDelAttr (0.00s)
	builtin_types_test.go:53: func1(<Foo object at 0xc420425f20>, 'bar') raised AttributeError('\'Foo\' object has no attribute \'bar\'',), want <nil>
Exception: foo
RuntimeError: foo
FAIL
FAIL	grumpy	1.197s
make: *** [build/runtime.pass] Error 1
```